### ISSUE TITLE
Pre-release workflow issue with v16-beta

### ISSUE THREAD

**Issue by hershmire** [semantic-release/semantic-release#1364]
## Current behavior

As I'm following along the defined [pre-release workflow steps](https://github.com/semantic-release/semantic-release/blob/v16.0.0-beta.31%40beta/docs/recipes/pre-releases.md) to test out my CI workflow I've run across an issue during the step where I [publish my beta release to the default distribution channel (`master`)](https://github.com/semantic-release/semantic-release/blob/v16.0.0-beta.31%40beta/docs/recipes/pre-releases.md#publishing-the-300-beta-release-to-the-default-distribution-channel). Once I merge `beta` to `master` and push changes up, a Github Action workflow is kicked off to run semantic-release. Unfortunately, it fails to finish with the following error:

```sh
EINVALIDNEXTVERSION The release `5.0.0` on branch `master` cannot be published as it is out of range.
```

I've posted the full semantic-release logs to better illustrate what is happening. Not sure if I did something incorrectly, the docs are incorrect, or there's an issue with `semantic-release@v16-beta`.

**NOTE:** This is on a private repo so I can't link you to that. If absolutely needed, I can create a public repo to try and simulate this issue. Here are some screenshots for more info:

**List of current branches in Github:**
<img width="288" alt="Screen Shot 2019-11-22 at 13 08 31" src="https://user-images.githubusercontent.com/757524/69457056-6f478600-0d29-11ea-882d-f8fb629b8e1e.png">

**List of current tags in Github:**
<img width="288" alt="Screen Shot 2019-11-22 at 13 08 42" src="https://user-images.githubusercontent.com/757524/69457080-7e2e3880-0d29-11ea-85e8-5830d76373b7.png">

**List of current releases in NPM:**
<img width="754" alt="Screen Shot 2019-11-22 at 13 09 13" src="https://user-images.githubusercontent.com/757524/69457114-93a36280-0d29-11ea-8a70-f7d93651329c.png">

## Expected behavior

Expected once `beta` was merged into `master` and pushed, semantic-release would successfully release a new major.

## Environment

**semantic-release** version: ^16.0.0-beta

CI environment: Github Actions using `ubuntu-latest`

Plugins used: See package.json config below 👇 

**semantic-release** configuration:

In package.json:
```json
"release": {
    "plugins": [
      "@semantic-release/commit-analyzer",
      "@semantic-release/release-notes-generator",
      "@semantic-release/changelog",
      "@semantic-release/npm",
      "@semantic-release/git"
    ]
  },
```

I'm using the following Github Actions for this:
```yaml
- name: Semantic Release
  id: semantic # Need an `id` for output variables
  uses: cycjimmy/semantic-release-action@v2
  with:
    branch: master
    semantic_version: ^16.0.0-beta
    extra_plugins: |
      @semantic-release/git@beta
      @semantic-release/changelog
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
    NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

CI logs:
```sh
[7:42:03 PM] [semantic-release] â€º â„¹  Running semantic-release version 16.0.0-beta.35
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/changelog"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/git"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "prepare" from "@semantic-release/changelog"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "prepare" from "@semantic-release/npm"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "prepare" from "@semantic-release/git"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "publish" from "@semantic-release/npm"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "addChannel" from "@semantic-release/npm"
[7:42:04 PM] [semantic-release] â€º âœ”  Run automated release from branch master
[7:42:05 PM] [semantic-release] â€º âœ”  Allowed to push to the Git repository
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "verifyConditions" of plugin "@semantic-release/changelog"
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "verifyConditions" of plugin "@semantic-release/changelog"
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "verifyConditions" of plugin "@semantic-release/npm"
[7:42:05 PM] [semantic-release] [@semantic-release/npm] â€º â„¹  Verify authentication for registry https://registry.npmjs.org/
[7:42:05 PM] [semantic-release] [@semantic-release/npm] â€º â„¹  Reading npm config from /home/runner/work/_temp/.npmrc
my-private-repo
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "verifyConditions" of plugin "@semantic-release/git"
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "verifyConditions" of plugin "@semantic-release/git"
[7:42:05 PM] [semantic-release] â€º â„¹  Found git tag v4.1.1 associated with version 4.1.1 on branch master
[7:42:05 PM] [semantic-release] â€º â„¹  Found 5 commits since last release
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: Merge branch 'beta'
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The commit should not trigger a release
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: chore(release): 5.0.0-beta.2 [skip ci]

# [5.0.0-beta.2](https://github.com/my-private-repo/test-singlerepo/compare/v5.0.0-beta.1@beta...v5.0.0-beta.2@beta) (2019-11-22)

### Features

* minor feature on [@beta](https://github.com/beta) ([5e890a2](https://github.com/my-private-repo/test-singlerepo/commit/5e890a27f1501c66d3eae54372eee27853d3b5b5))
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The commit should not trigger a release
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: feat: minor feature on @beta
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The release type for the commit is minor
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: chore(release): 5.0.0-beta.1 [skip ci]

# [5.0.0-beta.1](https://github.com/my-private-repo/test-singlerepo/compare/v4.1.0...v5.0.0-beta.1@beta) (2019-11-22)

### Features

* first feature on beta ([e0081b6](https://github.com/my-private-repo/test-singlerepo/commit/e0081b6144dd26ae1bd94cfb10cf88364e5125f0))

### BREAKING CHANGES

* it breaks something
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The commit should not trigger a release
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: feat: first feature on beta

BREAKING CHANGE: it breaks something
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The release type for the commit is major
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analysis of 5 commits complete: major release
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[7:42:06 PM] [semantic-release] â€º â„¹  The next release version is 5.0.0
[7:42:06 PM] [semantic-release] â€º âœ–  EINVALIDNEXTVERSION The release `5.0.0` on branch `master` cannot be published as it is out of range.
Based on the releases published on other branches, only versions within the range >=4.1.1 <5.0.0 can be published from branch master.

The following commits are responsible for the invalid release:

    * Merge branch 'beta' (1edc7a1)
    * chore(release): 5.0.0-beta.2 [skip ci] (32aeea1)
    * feat: minor feature on @beta (5e890a2)
    * chore(release): 5.0.0-beta.1 [skip ci] (6185875)
    * feat: first feature on beta (e0081b6)

Those commits should be moved to a valid branch with git merge (https://git-scm.com/docs/git-merge) or git cherry-pick (https://git-scm.com/docs/git-cherry-pick) and removed from branch master with git revert (https://git-scm.com/docs/git-revert) or git reset (https://git-scm.com/docs/git-reset).

A valid branch could be next.

See the workflow configuration documentation (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/workflow-configuration.md) for more details.

##[error]SemanticReleaseError: The release `5.0.0` on branch `master` cannot be published as it is out of range.
##[error]Node run failed with exit code 1
```

**Most recent commits from master:**
```sh
1edc7a11c67813bf158e641c331e63f046c1228f (HEAD -> master, origin/master, origin/HEAD) Merge branch 'beta'
7f9a4c4591850236843d6d5f77a672580c1b35f5 (tag: v4.1.1) chore(release): 4.1.1 [skip ci]
32aeea1201b15d5e9de6e24a2592793eb27c64b9 (tag: v5.0.0-beta.2@beta, origin/beta) chore(release): 5.0.0-beta.2 [skip ci]
31c01f4e045b43eeb292d6c721212992393f0537 fix: bug fix for v4.x.x
5e890a27f1501c66d3eae54372eee27853d3b5b5 feat: minor feature on @beta
6185875e16c33f8032a6d4049235fd70812e4b73 (tag: v5.0.0-beta.1@beta) chore(release): 5.0.0-beta.1 [skip ci]
e0081b6144dd26ae1bd94cfb10cf88364e5125f0 feat: first feature on beta
c44e232c34fe71e71101a835483da60a6ce41a8c (tag: v4.1.0@next, tag: v4.1.0, origin/next, next) chore(release): 4.1.0 [skip ci]
d780573d31810f86b37ebd3d3874a6e5fa6e5133 Merge branch 'master' into next
461bb393da2cfb097382391b60622aa031ea1786 (tag: v3.2.0) chore(release): 3.2.0 [skip ci]
44169c40c66930adb853fabf021bdc67c73f8876 feat: a small feat in latest 3.x.x
e6c4b10941efcb1631b670fe3b6f87297708b664 (tag: v4.0.1@next, tag: v4.0.1) chore(release): 4.0.1 [skip ci]
```

**Comment by pvdlg** [comment]
What tags do you have on the `next` branch?

**Comment by hershmire** [comment]
These are all my current tags:
```sh
v1.0.0
v1.0.1
v1.0.2
v1.0.3
v1.0.4
v1.1.0
v1.1.1
v1.1.2
v1.1.3
v1.1.4
v1.1.5
v1.1.6
v1.1.7
v1.1.8
v1.2.0
v1.2.1
v1.3.0
v2.0.0
v2.0.0@2.x
v2.1.0@2.x
v2.1.1@2.x
v3.0.0
v3.1.0
v3.1.1
v3.2.0
v4.0.0
v4.0.0@next
v4.0.1
v4.0.1@next
v4.1.0
v4.1.0@next
v4.1.1
v5.0.0-alpha.1@alpha
v5.0.0-beta.1@beta
v5.0.0-beta.2@beta
```

**Comment by pvdlg** [comment]
I would need to know which ones are in the history of `next`.
You can get that with `git tag --merged next`.

Could you also re-run with `--debug` option and provide the logs (in Gist or something like that if it's too long).

**Comment by hershmire** [comment]
Here you go:

```sh
git tag --merged next
v1.0.0
v1.0.1
v1.0.2
v1.0.3
v1.0.4
v1.1.0
v1.1.1
v1.1.2
v1.1.3
v1.1.4
v1.1.5
v1.1.6
v1.1.7
v1.1.8
v1.2.0
v1.2.1
v1.3.0
v2.0.0
v2.0.0@2.x
v2.1.0@2.x
v3.0.0
v3.1.0
v3.1.1
v3.2.0
v4.0.0
v4.0.0@next
v4.0.1
v4.0.1@next
v4.1.0
v4.1.0@next
```

I will try running with `--debug` and report back as well.

**Comment by pvdlg** [comment]
Actually I think it's working as intended. See https://github.com/semantic-release/semantic-release/blob/beta/docs/usage/workflow-configuration.md#pushing-to-a-release-branch

> With the configuration `"branches": ["master", "next"]`, if the last release published from `master` is `1.0.0` then:
> - Only versions in range `1.x.x` can be published from `master`, so only `fix` and `feat` commits can be pushed to `master`
> - Only versions in range `>=2.0.0` release can be published from `next`, so a `BREAKING CHANGE` commit must be pushed first on `next` and can be followed by any type of commits

So in your case you should merge `beta` into `next` first which is going to release `5.0.0` on `@next` and then merge `next` into `master` which will make `5.0.0` available on `@latest`.

Not sure if we should make an exception to this rule to allow to bypass  `next` when it is not ahead of `master`.

**Comment by hershmire** [comment]
Interesting, doing what you suggested worked for me. I didn't see that part of the docs. Thanks for the clarification!
